qcodes/instrument_drivers/signal_hound/USB_SA124B.py

```diff
@@ -1,166 +1,327 @@
+        self._trace_updated = False
+        log.info('Initializing instrument SignalHound USB 124B')
         self.dll = ct.CDLL(dll_path or self.dll_path)
         self.hf = constants
 
         self.add_parameter('frequency',
                            label='Frequency ',
                            unit='Hz',
                            initial_value=5e9,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers())
+                           vals=vals.Numbers(),
+                           parameter_class=SweepTraceParameter)
         self.add_parameter('span',
                            label='Span ',
                            unit='Hz',
                            initial_value=.25e6,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers())
+                           vals=vals.Numbers(),
+                           parameter_class=SweepTraceParameter)
+        self.add_parameter('npts',
+                           label='Number of Points',
+                           get_cmd=None,
+                           set_cmd=False)
+        self.add_parameter('avg',
+                           label='Averages',
+                           initial_value=1,
+                           get_cmd=None,
+                           set_cmd=None,
+                           vals=vals.Ints())
         self.add_parameter('power',
                            label='Power ',
                            unit='dBm',
-                           initial_value=0,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers(max_value=20))
+                           get_cmd=self._get_power_at_freq,
+                           set_cmd=False)
         self.add_parameter('ref_lvl',
                            label='Reference power ',
                            unit='dBm',
                            initial_value=0,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers(max_value=20))
+                           vals=vals.Numbers(max_value=20),
+                           parameter_class=TraceParameter)
         self.add_parameter('external_reference',
-                           get_cmd=None, set_cmd=None,
                            initial_value=False,
-                           vals=vals.Bool())
+                           vals=vals.Bool(),
+                           parameter_class=TraceParameter)
         self.add_parameter('device_type',
-                           get_cmd=self._do_get_device_type)
-
+                           set_cmd=False,
+                           get_cmd=self._get_device_type)
         self.add_parameter('device_mode',
-                           initial_value='sweeping',
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Anything())
+                           get_cmd=lambda: 'sweeping',
```

Comment:
Yes @nataliejpg effectively made this a stub because the other modes are not implemented. It used to be possible to change this but the driver would silently change it back. I think this is ok as an entry point for any one who wants to add this functionality in the future but i will document the intention.